### PR TITLE
mat2: init at 0.11.0

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Adds `mat2`, a program designed to remove metadata information from a variety of file types. This can be useful for not accidentally revealing personal information when sending files.

Upstream is at: https://0xacab.org/jvoisin/mat2/

`doCheck = false` was set, as there was an issue with internal library detection during the check phase. If there is a known solution to this, please let me know.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1694,6 +1694,16 @@
     githubId = 4331004;
     name = "Naoya Hatta";
   };
+  Dallos = {
```
Comment (alexarice) [inline maintainers/maintainer-list.nix:1697]: The addition to `maintainer-list.nix` should be in a separate commit

File: pkgs/tools/graphics/mat2/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{ stdenv
```
Comment (alexarice) [inline pkgs/tools/graphics/mat2/default.nix:1]: You could import `lib` instead of `stdenv`

File: pkgs/tools/graphics/mat2/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    pygobject3
+    librsvg
+    ffmpeg
+    exiftool
+    gobject-introspection
+    poppler_gi
+    gdk-pixbuf
+  ];
+
+  meta = {
```
Comment (alexarice) [inline pkgs/tools/graphics/mat2/default.nix:39]: Could put `with lib;` here

File: pkgs/tools/graphics/mat2/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+in buildPythonApplication rec {
+  pname = "mat2";
+  version = "0.11.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "beb99965eb41ebe022b6abed941aa0d23882e26b4b4b48f94295911a637f90ff";
+  };
+
+  # avoids AttributeError: module 'libmat2.archive' has no attribute 'ZipParser'
```
Comment (alexarice) [inline pkgs/tools/graphics/mat2/default.nix:23]: Is there an upstream issue for this?

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -3123,6 +3123,8 @@ in
 
   magic-vlsi = callPackage ../applications/science/electronics/magic-vlsi { };
 
+  mat2 = callPackage ../tools/graphics/mat2 {};
```
Comment (alexarice) [inline pkgs/top-level/all-packages.nix:3126]: Graphics seems like a strange choice, this works for a variety of file types right?

Comment (Dallos) [reply pkgs/top-level/all-packages.nix:3126]: Yes, it works with many multimedia types. I'll move it to misc.

File: pkgs/tools/misc/mat2/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+in buildPythonApplication rec {
+  pname = "mat2";
+  version = "0.11.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "beb99965eb41ebe022b6abed941aa0d23882e26b4b4b48f94295911a637f90ff";
+  };
+
+  # avoids AttributeError: module 'libmat2.archive' has no attribute 'ZipParser'
```
Comment (ryantm) [inline pkgs/tools/misc/mat2/default.nix:23]: Please report upstream and add comment linking to report

File: pkgs/tools/misc/mat2/default.nix [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    exiftool
+    gobject-introspection
+    poppler_gi
+    gdk-pixbuf
+  ];
+
+  meta = with lib; {
+    description = "metadata removal tool supporting a wide range of commonly used file formats";
+    homepage = "https://0xacab.org/jvoisin/mat2";
+    license = licenses.gpl3;
```
Comment (ryantm) [inline pkgs/tools/misc/mat2/default.nix:42]: The gpl license variants without the `Plus` or `Only` suffix are deprecated, because they do not specify the license well enough. Use `Plus` if the project has "or later" sections in the actual source code files (not the license). Otherwise, use the `Only` suffix.
 
See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

**Review by ryantm — CHANGES_REQUESTED** [review]

**Comment by ryantm** [comment]
Result of `nixpkgs-review pr 90160` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- mat2
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 90160` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mat2</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Closing because the author is a ghost.
